awscli/clidocs.py

```diff
@@ -159,6 +161,14 @@ def doc_options_start(self, help_command, **kwargs):
     def doc_option(self, arg_name, help_command, **kwargs):
         doc = help_command.doc
         argument = help_command.arg_table[arg_name]
+        if is_streaming_blob_type(argument.argument_model):
+            doc.style.start_note()
+            msg = ("This argument is of type: streaming blob. "
+                   "Its value must be the path to a file "
+                   "(e.g. ``path/to/file``) and must **not** "
+                   "be prefixed with ``file://`` or ``fileb://``")
+            doc.include_doc_string(msg)
+            doc.style.end_note()
```

Comment:
Let's encapsulate this logic as a `_add_streaming_blob_note()` method. The `doc_option()` method is already too long and could use some refactoring. So this will help ensure it does not make readability any worse.